src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs

```diff
@@ -120,60 +148,86 @@ private SafeCreateHandle PreparePoliciesArray(bool checkRevocation)
             return policiesArray;
         }
 
-        private SafeCreateHandle PrepareCertsArray(ICertificatePal cert, X509Certificate2Collection extraStore)
+        private SafeCreateHandle PrepareCertsArray(
+            ICertificatePal cert,
+            X509Certificate2Collection extraStore,
+            X509Certificate2Collection customTrustStore,
+            X509ChainTrustMode trustMode)
         {
-            IntPtr[] ptrs = new IntPtr[1 + (extraStore?.Count ?? 0)];
-            SafeHandle[] safeHandles = new SafeHandle[ptrs.Length];
-
+            List<SafeHandle> safeHandles = new List<SafeHandle>();
             AppleCertificatePal applePal = (AppleCertificatePal)cert;
-
-            safeHandles[0] = applePal.CertificateHandle;
+            safeHandles.Add(applePal.CertificateHandle);
 
             if (extraStore != null)
             {
                 for (int i = 0; i < extraStore.Count; i++)
                 {
                     AppleCertificatePal extraCertPal = (AppleCertificatePal)extraStore[i].Pal;
+                    safeHandles.Add(extraCertPal.CertificateHandle);
+                }
+            }
+
+            if (trustMode == X509ChainTrustMode.CustomRootTrust && customTrustStore != null)
+            {
+                for (int i = 0; i < customTrustStore.Count; i++)
+                {
+                    // Only adds non self issued certs to the untrusted certs array. Trusted self signed
+                    // certs will be added to the custom certs array.
+                    if (!cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData))
```

Comment:
Why is this `if` inside the `for` loop rather than outside of it?

Reply:
Hm. I think it's supposed to be using `customTrustStore[i]` instead of cert.  Good eye.